### PR TITLE
STUD-782 Added WK CI

### PR DESCRIPTION
**Pull request by joecotton-wk**
## Purpose

Adding WK CI so that it builds correctly

## Solution

High level overview of what was done. This is the roadmap for those who are going to CR.

## Semantic Versioning (check one)

- [ ] The following were changed in a non-backward compatible way and requires a major version bump:
  - _[link to the breaking change in the diff]_
- [ ] Something public was added or changed in backward compatible way, this requires a minor version bump
- [x] No public changes nor new features (backwards-compatible refactor or bug fix), so this can be included in a patch
      release

## How to QA

- [ ] All relevant references to `OneCloudInc` have been replaced with `Workiva`
- [ ] `go build ./...` runs successfully
- [ ] Build passes
- [ ] Skynet passes
- [ ] Aviary passes
- [ ] Rosie runs her checks

### CODE REVIEW

**Comment by aviary-wf** [comment]
## Security Insights
The items listed below may not capture all security relevant changes. Before providing a security review, be sure to review the entire PR for security impact.
<details>
<summary>(1) Security relevant changes were detected</summary>
<li>Watched file <code>aviary.yaml</code> added</li>
</details>

## Action Items
* Request infosec review via Slack: #support-infosec
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: Dockerfile [hunk 1]
```diff
@@ -1,14 +1,3 @@
...
-WORKDIR /go/src/app
-COPY . .
-
-RUN go get -d -v ./...
-RUN go install -v ./...
-
-WORKDIR /tmp
-
-ENTRYPOINT ["textql"]
+# This is a Golang library; it doesn't produce
```
Comment (andrewrowe-wk) [inline Dockerfile:1]: This actually does produce an executable that we run in Droplet.

Comment (joecotton-wk) [reply Dockerfile:1]: oh man, I was just doing this one like all of the rest of them 🤦‍♂️ .

So, this produces a Docker container which is consumed by droplet? I searched the droplet repo for `textql`, and didn't get a single hit.

Comment (andrewrowe-wk) [reply Dockerfile:1]: Good point... The executable created by the build process, which is not super easy to make, runs directly on the OS, and not the Docker container. I wonder if @brianmyers-wf can comment on this.

File: Makefile [hunk 2]
```diff
@@ -1,25 +1,10 @@
-.PHONY: all test clean man godep fast release install
+SHELL := /bin/bash
 
-all: textql
+test: tools
```
Comment (andrewrowe-wk) [inline Makefile:3]: This is probably fine... Why did you remove all the other Make processes?

Comment (joecotton-wk) [reply Makefile:3]: I have a script that applies a certain set of changes to these repos. I just ran it and didn't think anything of it, considering none of the others had makefiles 🤦‍♂️ . Will restore.

File: test-reports/unit-test-report.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (joecotton-wk) [inline test-reports/unit-test-report.xml:1]: This shouldn't be there...going to remove it

Comment (joecotton-wk) [reply test-reports/unit-test-report.xml:1]: This is explicitly ignored in the `.gitignore`....not sure how it ended up here 🤷‍♂️

File: go.mod [hunk 4]
```diff
@@ -1,9 +1,9 @@
-module github.com/dinedal/textql
+module github.com/Workiva/textql
 
 go 1.13
 
 require (
-	github.com/mattn/go-runewidth v0.0.2 // indirect
+	github.com/dinedal/textql v0.0.0-20210706151746-fca00ecc76c8
```
Comment (andrewrowe-wk) [inline go.mod:6]: Also, does it not need the runewidth package anymore?

File: Dockerfile (hunk 1 above)
Comment (brianmyers-wf) [reply Dockerfile:1]: Sorry, I'm unfamiliar with anything going on here... perhaps you've meant to at someone else?

File: go.mod (hunk 4 above)
Comment (joecotton-wk) [reply go.mod:6]: I ran `go mod tidy` and `go mod tidy` removed it. So i guess not.

FWIW I did a search and didn't see it used in an import statement anywhere. The only usage is in the Gopgk.lock file, but idk if that's used.

File: Dockerfile (hunk 1 above)
Comment (joecotton-wk) [reply Dockerfile:1]: haha, i think you're right...we meant to @brianmeyers-wk 😆 sorry for the interruption!

Comment (andrewrowe-wk) [reply Dockerfile:1]: lol

**Comment by joecotton-wk** [comment]
So, after discussing this further with @andrewrowe-wk:

- This repo never had a build pipeline on the OC side. If it did, it was broken and useless.
- This repo hasn't been built since Feb 2020, and it had to be built locally on someone's machine due to complexities in the build process
- To our knowledge, this repo isn't used as a direct import by any other repo (even tho its compiled artifacts are used in both the plugin and in the `onecloud-csv` bizapp in cerebral)

Our primary purpose in migrating repositories is to leave them in the exact same state they were in in the OneCloudInc Github org. No pipelines existed there, and so it doesn't make sense to add them now. Additionally, adding them now would force us to cross a bunch of bridges that previous devs haven't crossed yet. It doesn't make sense to add these now (and it may not ever).

Closing this without the intention of reopening.
